src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildInfoAction.java

```diff
@@ -0,0 +1,65 @@
+package org.jenkinsci.plugins.workflow.support.steps.build;
+
+import hudson.model.Job;
+import hudson.model.Run;
+import jenkins.model.Jenkins;
+import hudson.model.InvisibleAction;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+
+public class BuildInfoAction extends InvisibleAction {
+    private final Queue<BuildInfo> buildInfos = new ConcurrentLinkedQueue<>();
+
+    BuildInfoAction() {
+    }
+
+    BuildInfoAction(String projectName, int buildNumber) {
+        buildInfos.add(new BuildInfo(projectName, buildNumber));
+    }
+
+    public List<Run<?, ?>> getChildBuilds() {
+        if (buildInfos.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<Run<?, ?>> builds = new ArrayList<>(buildInfos.size());
+
+        for (BuildInfo buildInfo : buildInfos) {
+            if (buildInfo != null) {
+                Job job = Jenkins.getActiveInstance().getItemByFullName(buildInfo.getProjectName(), Job.class);
+                if (buildInfo.getBuildNumber() != 0) {
+                    builds.add((job != null) ? job.getBuildByNumber(buildInfo.getBuildNumber()) : null);
+                }
+            }
+        }
+
+        return builds;
+    }
+
+    public void addBuildInfo(String projectName, int buildNumber) {
+        buildInfos.add(new BuildInfo(projectName, buildNumber));
+    }
+
+    private static class BuildInfo {
+        private final String projectName;
+        private final int buildNumber;
```

Comment:
It does not look easier for me, in this case I would need one String concatenation + `String.split` for every call for nothing. I now that it's tiny stuff, but even from code perspective it doesn't simplify much.